react/components/DatePicker/index.js

```diff
@@ -19,6 +19,43 @@ class DatePicker extends Component {
     this.handleLocaleChange(props.locale)
   }
 
+  get popperModifiers() {
+    const { align, direction, size, useTime } = this.props
+
+    const isRightAligned = align === 'right'
+    const isUpwardsDirected = direction === 'up'
+
+    if (!isRightAligned && !isUpwardsDirected) {
+      return undefined
+    }
+
+    const offsetX = isRightAligned
+      ? (size === 'large' ? -91 : -136) - (useTime ? 130 : 0)
+      : 0
+
+    const offsetYBySize = {
+      large: -448,
```

Comment:
these too. are they always fixed regardless of CSS?

Reply:
Also observation. I'd say yes, because all that matters is the size of the input. Can you think of an edge case that I might've missed? Perhaps size transformations?